File: 387-FirstUnique.py
def firstUniqChar(s): 
    dict_char = {}
    i = 0
    
    while i < len(s): 
        if s[i] in dict_char: 
            dict_char[s[i]] += 1
        else: 
            dict_char[s[i]] = 1
        i += 1
    for key in dict_char: 
        if dict_char[key] == 1:
            return s.index(key)
    return -1

print(firstUniqChar("leetcode"))

File: 387-FirstUnique.js
var firstUniqChar = function(s) {
    let numObj = {};
    for (let i = 0; i <= s.length - 1; i++) {
        if(s[i] in numObj) {
            numObj[s[i]]++;
        } else {
            numObj[s[i]] = 1;
        }
    }
    for (let key in numObj) {
        if(numObj[key] === 1) {
            return s.indexOf(key);
        }
    }
    return -1

};

console.log(firstUniqChar("leetcode"))
